renderer/shaders/advanced_blend.glsl

```diff
@@ -172,14 +172,14 @@ half4 advanced_blend(half4 src, half4 dst, ushort mode)
         case BLEND_MODE_COLORDODGE:
             // ES3 spec, 4.5.1 Range and Precision: dividing a non-zero by 0 results in the
             // appropriately signed IEEE Inf.
-            f = mix(min(dst.rgb / (1. - src.rgb), make_half3(1, 1, 1)),
+            f = (half)mix(min(dst.rgb / (1. - src.rgb), make_half3(1, 1, 1)),
```

Comment:
Unfortunately, a C-style `(half)` cast won't work in all the languages we support. We have a `make_half()` macro we use instead.

Also, is `mix()` overloaded? What if we instead do:

```
f = mix(min(dst.rgb / (make_half(1) - src.rgb), make_half3(1, 1, 1)),
        make_half3(0, 0, 0),
        lessThanEqual(dst.rgb, make_half3(0, 0, 0)));

```

Do we hit a `half` overload of `mix()` when we explicitly make the 1 a half?

Reply:
Yeah, I still get the conversion warning with your example. In this context, `mix` is defined as `lerp`, which looks like it only returns `float`.

Reply:
Oh right, this is custom defined in `hlsl.glsl`

Can we instead define overloads of `mix` that accept both `half` and `float`? The float ones can use lerp and the half ones can use `x + s*(y-x).`

Reply:
Sure thing, I can set that up. We would need overloads for each of the vector types as well (`half2`/`float2`/`half3`/etc.)
Do you have a preference of whether I handle that using macros, or just manually write out each function separately? Happy to do either, just don't want to go too far into macro magic if you'd prefer to avoid it ;)

Reply:
Thanks Elliot!

Since these can be one liners, I think I'd prefer the least amount of extra macro magic possible  😁